### ISSUE TITLE
ALT key sticking when ALT-TAB out of the VNC session

### ISSUE THREAD

**Issue by calisro** [FreeRDP/Remmina#1082]
We support Remmina 1.2+ versions only.
Please do not report bugs occurring on older versions!

Please give us some details about the systems you are using:
* Client (OS name and version): 
Ubuntu connecting to x11vnc
* Remmina version (remmina --version): 
remmina - Version 1.2.0-rcgit-17 (git n/a)
* Desktop environment (GNOME, Unity, KDE, ..): 
XFCE4
* Connecting to (OS and version): 
Ubuntu connecting to x11vnc
* Connecting via (RDP, VNC, ...): 
VNC
* RDP issues only, freerdp version (xfreerdp --version): 

When alt-tabbing out of a VNC session, it seems to stick the ALT key on the VNC session.  When i click back into that session the ALT key seems stuck.  I can then click the ALT key once to 'free' it and things work fine but this really affects the functionaliy.  It seems to happen ONLY when the VNC is focused and I ALT+TAB out of the VNC to one of my desktop applications on the client side.

**Comment by antenore** [comment]
I cannot reproduce it, what VNC server are you using?

**Comment by calisro** [comment]
I'm using 
x11vnc -version
x11vnc: 0.9.13 lastmod: 2011-08-10

**Comment by calisro** [comment]
Would it help if I try to recreate this using a different server?  Do you suspect the server itself rather than the client?

**Comment by antenore** [comment]
I'll try x11vnc as well, but if you can test also another server it'd be great.

**Comment by calisro** [comment]
I tried TigerVNC server and I encountered the same problem immediately.  
https://bintray.com/tigervnc/stable/tigervnc/1.7.0#files/ubuntu-12.04LTS/amd64

**Comment by attilastrba** [comment]
I see the same issue with rdp. This happens since I have upgraded  Version 1.2.0-rcgit-15.

**Comment by attilastrba** [comment]
Ok so it is not a remmina issue. This feature is not implemented in Wayland... read here further infos https://bugzilla.redhat.com/show_bug.cgi?id=1285770

**Comment by antenore** [comment]
@attilastrba good! Thanks for this, I don't use Wayland (yet).
I keep this opened to keep track.

**Comment by attilastrba** [comment]
Just one more minor note as a workaround:  changing to XORG seems to solve the issue. 
For the fedora users just edit /etc/gdm/custom.conf and set WaylandEnable=false

**Comment by calisro** [comment]
Additionally, I was using older version of remmina and never had this issue on xfce4.  It was only on upgrade to remminas current version tjis issue came.

**Comment by antenore** [comment]
> Additionally, I was using older version of remmina and never had this issue on xfce4. It was only on upgrade to remminas current version tjis issue came.

This is because we added support for Wayland.

When you start Remmina, we check which back-end is available, as per this code https://github.com/FreeRDP/Remmina/blob/next/remmina/src/remmina.c#L228

To workaround the back-end "chosen" by us/GTK, you can simply set the back-end manually, like: 

```sh
GDK_BACKEND=x11 remmina

```
For more information read https://developer.gnome.org/gtk3/stable/gtk-running and if you want to have some fun, try the broadway back-end in a web browser.

https://wiki.archlinux.org/index.php/GTK%2B#Broadway_backend

> Just one more minor note as a workaround: changing to XORG seems to solve the issue.
For the fedora users just edit /etc/gdm/custom.conf and set WaylandEnable=false 

@attilastrba this is not needed luckily

**Comment by antenore** [comment]
BTW 2 notes

1. I'm probably going to modify the title of the issue.
2. I'm not sure if the GDK_BACKEND will work with XWayland, probably you need to have an xorg installed and configured... Probably.

**Comment by calisro** [comment]
I tried launching it with GDK_BACKEND=x11 remmina and I still encounter the issue.

**Comment by antenore** [comment]
@calisro yes, sorry I was mixing up the two issues as you already have Xorg, it's a libvnc or Remmina issue. We are invetigating @attilastrba can you please an other issue for Wayland ?

**Comment by antenore** [comment]
I can replicate your issue only if I *don't set* "Map Meta Keys" for "Keyboard mapping" (under basic setting for each profile).

Can you see what you have and make some tests?

**Comment by calisro** [comment]
I originally had them 'unset'.  I tried to both set them and then unset them testing both set and unset and I can recreate the issue for both cases consistently.

**Comment by calisro** [comment]
This is exactly what I am doing.

I open a VNC remmina.  I click into the session.  I alt-tab to a different window on my client.  I click back with the mouse into the remmina session.  The alt key is now stuck.

**Comment by antenore** [comment]
I'm buffed... Which Ubuntu version are you using?

**Comment by calisro** [comment]
└─>  lsb_release -a 
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty

**Comment by calisro** [comment]
I used the original remmina for years without issue.  I recently upgraded remmina because it seemed to be pretty old.  I've never had key capture problems like this with the alt key.  Is there any debug or anything else I can do?  Its hard to do anything when you can't recreate it.

**Comment by attilastrba** [comment]
@antenore I have created a new ticket according Wayland [1086](https://github.com/FreeRDP/Remmina/issues/1086)
If I understand correctly the mentioned workaround should work on Wayland if I have Xorg right? I will test it

**Comment by antenore** [comment]
@attilastrba yes, right, thanks for your precious support

**Comment by attilastrba** [comment]
thank you for maintaining remmina, I couldn't live without it ;)

so I tried it with Wayland and the GDK_BACKEND=x11 but did not work, neither rdp or vnc
I got some warnings and errors I don't know if relevant but pasting it:

```
[10:52:24] attila@attila-thinkpad /home/attila (0) 
> loginctl show-session 2 -p Type
Type=wayland
[10:52:26] attila@attila-thinkpad /home/attila (0) 
> bash
[attila@attila-thinkpad ~]$ GDK_BACKEND=x11 remmina
Failed to load plugin: /usr/lib64/remmina/plugins/remmina-plugin-telepathy.so.
Error: /usr/lib64/remmina/plugins/remmina-plugin-telepathy.so: undefined symbol: remmina_tp_handler_new

(remmina:15144): Gtk-WARNING **: gtk_menu_attach_to_widget(): menu already attached to GtkMenuItem
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
[10:52:34:977] [15144:15148] [INFO][com.freerdp.client.common.cmdline] - loading channel cliprdr

(remmina:15144): Gtk-WARNING **: Theme parsing error: <data>:16:28: The style property GtkWidget:focus-padding is deprecated and shouldn't be used anymore. It will be removed in a future version

(remmina:15144): Gtk-WARNING **: Theme parsing error: <data>:17:31: The style property GtkWidget:focus-line-width is deprecated and shouldn't be used anymore. It will be removed in a future version

(remmina:15144): Gtk-WARNING **: Theme parsing error: <data>:22:28: The style property GtkWidget:focus-padding is deprecated and shouldn't be used anymore. It will be removed in a future version

(remmina:15144): Gtk-WARNING **: Theme parsing error: <data>:23:31: The style property GtkWidget:focus-line-width is deprecated and shouldn't be used anymore. It will be removed in a future version

(remmina:15144): Gtk-WARNING **: Allocating size to remmina-cw-viewport 0x55cd1e3d06f0 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?
```

**Comment by antenore** [comment]
Thanks :-)

I'm afraid we have to wait for a solution from the FreeDesktop group

**Comment by chiamarc** [comment]
I can confirm the same exact behavior with RDP to a Windows box for a couple years now from Fedora 19 up through Fedora 23.  I am also unable to get the workaround to function.  If it helps, the same behavior obtains when I use the NoMachine client from my Fedora desktop to my Windows laptop!

**Comment by jrmrjnck** [comment]
I can confirm this issue, and it makes Remmina unusable for me. Every time I switch back to the VNC window, the Alt key is stuck and I trigger all sorts of random actions. No wayland on client or server.

**Comment by antenore** [comment]
WARNING:
The Remmina project is moving to gitlab.com, in order to keep receiving notification about this issue,
you should have a gitlab account and link it to github.
The migration will happen in the next days.
Thanks for your support and understanding.

cc @antenore @attilastrba @calisro @chiamarc @jrmrjnck
